cartography/intel/azure/key_vaults.py

```diff
@@ -178,77 +217,106 @@ def _load_key_vault_secrets_tx(
 ) -> None:
 
     ingest_secrets = """
-    UNWIND {SECRETS} as sec
+    UNWIND $SECRETS as sec
     MERGE (s:AzureKeyVaultSecret{id: sec.id})
     ON CREATE SET s.firstseen = timestamp(),
-    s.id = sec.id,
+    s.name = sec.name,
+    s.type = sec.type,
+    s.location = sec.location,
+    s.region = sec.location,
+    s.secret_uri = sec.properties.secret_uri,
+    s.enabled = sec.properties.attributes.enabled,
+    s.secret_uri_with_version = sec.properties.secret_uri_with_version,
+    s.created = sec.properties.attributes.created,
+    s.updated = sec.properties.attributes.updated,
     s.consolelink = sec.consolelink,
     s.contentType = sec.content_type,
     s.managed = s.managed
-    SET s.lasupdated = {update_tag}
+    SET s.lasupdated = $update_tag
     WITH s, sec
     MATCH (keyvault:AzureKeyVault{id: sec.vault_id})
     MERGE (keyvault)-[r:HAS_SECRET]->(s)
     ON CREATE SET r.firstseen = timestamp()
-    SET r.lastupdated = {update_tag}
+    SET r.lastupdated = $update_tag
     """
 
     tx.run(
         ingest_secrets,
-        SECRETS = key_vault_secrets,
+        SECRETS=key_vault_secrets,
         SUBSCRIPTION_ID=subscription_id,
         update_tag=update_tag,
     )
 
-@timeit
-def get_key_vault_certificates(client: CertificateClient, vault_id: str, regions: List, common_job_parameters: Dict) -> List[Dict]:
 
+@timeit
+def get_key_vault_certificates(client: CertificateClient, vault: Dict) -> List[Dict]:
+    certificates_list = []
     try:
-        certificates_list = list(map(lambda x:x.as_dict(), client.list_properties_of_certificates()))
+        certificates_data = client.list_properties_of_certificates()
+        for certificate in certificates_data:
+            certificate_data = {}
+            certificate_data['created_on'] = certificate.created_on
+            certificate_data['enabled'] = certificate.enabled
+            certificate_data['expires_on'] = certificate.expires_on
+            certificate_data['id'] = certificate.id
+            certificate_data['name'] = certificate.name
+            certificate_data['not_before'] = certificate.not_before
+            certificate_data['updated_on'] = certificate.updated_on
+            certificate_data['version'] = certificate.version
+            certificates_list.append(certificate_data)
         return certificates_list
-    except HttpResponseError as e:
+    except Exception as e:
```

Comment:
We should not handle Exception (root class) in our code. Is HttpResponseError not able to handle the exception that you are looking for?